oauth2_client/migrations/0003_auto_20191129_0253.py

```diff
@@ -0,0 +1,146 @@
+            field=models.CharField(choices=[('authorization-code', 'Authorization code'), ('implicit', 'Implicit'), ('password', 'Resource owner password-based'), ('client-credentials', 'Client credentials'), ('jwt-bearer', 'JWT bearer')], default='client-credentials', max_length=32),
+            preserve_default=False,
+        ),
+        migrations.AddField(
+            model_name='application',
+            name='client_type',
+            field=models.CharField(choices=[('confidential', 'Confidential'), ('public', 'Public')], default='public', max_length=32),
+            preserve_default=False,
+        ),
+        migrations.AddField(
+            model_name='application',
+            name='extra_settings',
+            field=django.contrib.postgres.fields.jsonb.JSONField(blank=True, default=dict),
+        ),
+        migrations.AddField(
+            model_name='application',
+            name='provider',
+            field=models.CharField(default='labster', max_length=30, verbose_name='provider'),
+            preserve_default=False,
+        ),
+        migrations.AddField(
+            model_name='application',
+            name='token_uri',
+            field=models.CharField(blank=True, help_text='Token URI. Called to fetch the access token. For example: http://service-a:8000/o/token/', max_length=200),
+        ),
+        migrations.AddField(
+            model_name='application',
+            name='user',
+            field=models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='oauth2_client_application', to=settings.AUTH_USER_MODEL),
+        ),
+        migrations.AlterField(
+            model_name='accesstoken',
+            name='application',
+            field=models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='oauth2_client.Application'),
+        ),
+        migrations.AlterField(
+            model_name='accesstoken',
+            name='id',
+            field=models.BigAutoField(primary_key=True, serialize=False),
+        ),
+        migrations.AlterField(
+            model_name='accesstoken',
+            name='token',
+            field=models.CharField(max_length=255, unique=True),
+        ),
+        migrations.AlterField(
+            model_name='application',
+            name='client_id',
+            field=models.CharField(max_length=100),
+        ),
+        migrations.AlterField(
+            model_name='application',
+            name='client_secret',
+            field=models.CharField(blank=True, max_length=255),
+        ),
+        migrations.AlterField(
+            model_name='application',
+            name='id',
+            field=models.BigAutoField(primary_key=True, serialize=False),
+        ),
+        migrations.AlterField(
+            model_name='application',
+            name='name',
+            field=models.CharField(max_length=255, unique=True),
+        ),
+    ]
```

Comment:
how those changes are going to be rolled out to existing installations where data exists?

Reply:
I thought Application table is populated during every release with release scripts?

Reply:
> I thought Application table is populated during every release with release scripts?

it's only happen during sandbox creation.

Reply:
hmmm... pull the current prod data and prepare a one-time migration script? do we do that here at Labster?

Reply:
> hmmm... pull the current prod data and prepare a one-time migration script? do we do that here at Labster?

Please check whether data migration helps https://docs.djangoproject.com/en/3.0/topics/migrations/#data-migrations

Reply:
yes, this should do. Will prepare a migration script.

Reply:
not relevant anymore as the model is simplified now